mesonbuild/interpreter/type_checking.py

```diff
@@ -376,6 +377,13 @@ def _output_validator(outputs: T.List[str]) -> T.Optional[str]:
     default=[],
 )
 
+def include_dir_string_new(val: T.List[T.Union[str, IncludeDirs]], subproject: SubProject, node: mparser.BaseNode) -> None:
+    strs = [v for v in val if isinstance(v, str)]
+    if strs:
+        str_msg = ", ".join(f"'{s}'" for s in strs)
```

Comment:
```suggestion
        str_msg = ", ".join(repr(s) for s in strs)
```

Reply:
`repr()` can return either `"str"` or `'str'`, because `repr()` returns a value that is valid when passed to `eval()`, in meson only `'str'` is valid so we should use that.

Reply:
repr "can" do lots of things including the pattern frequently used in Meson's internal objects, which is to return
```
<str: contents>
```

But python uses single quotes for reprs here...

Reply:
Yes, we misuse the `__repr__` method, we should really knock that off.

CPython happens to do what we want right now, but there is no guarantee that it will continue to work, the only guarantee we have is that `eval(repr('foo')) == "foo"` is true. "It happens to work" is a pretty bad reason to do something, we're relying on behavior that python makes no guarantee of, and we should stop that.